src/post/like.rs

```diff
@@ -1,12 +1,14 @@
 use crate::str_serializers::*;
-use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::borsh::{BorshDeserialize, BorshSerialize};
 use near_sdk::serde::{Deserialize, Serialize};
 use near_sdk::{AccountId, Timestamp};
 use std::cmp::Ordering;
 use std::hash::{Hash, Hasher};
 
 #[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
 #[serde(crate = "near_sdk::serde")]
+#[borsh(crate = "near_sdk::borsh")]
+#[derive(Ord)]
```

Comment:
Can you shed some light on why Ord is needed here now? (If needed, let's merge it into the list of derives above

Reply:
Without Ord there is an error:
```
error[E0277]: the trait bound `like::Like: Ord` is not satisfied
  --> src/post/mod.rs:71:10
   |
71 | #[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
   |          ^^^^^^^^^^^^^^ the trait `Ord` is not implemented for `like::Like`
   |
   = help: the trait `near_sdk::borsh::BorshSerialize` is implemented for `HashSet<T, H>`
   = note: required for `HashSet<like::Like>` to implement `near_sdk::borsh::BorshSerialize`
   = note: this error originates in the derive macro `BorshSerialize` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `like::Like` with `#[derive(Ord)]`
```
For this struct:
```
#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
#[serde(crate = "near_sdk::serde")]
#[borsh(crate = "near_sdk::borsh")]
pub struct Post {
    pub id: PostId,
    pub author_id: AccountId,
    pub likes: HashSet<Like>,
    pub snapshot: PostSnapshot,
    // Excludes the current snapshot itself.
    pub snapshot_history: Vec<PostSnapshot>,
}
```

Reply:
Without Ord there is an error:
```
error[E0277]: the trait bound `like::Like: Ord` is not satisfied
  --> src/post/mod.rs:71:10
   |
71 | #[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
   |          ^^^^^^^^^^^^^^ the trait `Ord` is not implemented for `like::Like`
   |
   = help: the trait `near_sdk::borsh::BorshSerialize` is implemented for `HashSet<T, H>`
   = note: required for `HashSet<like::Like>` to implement `near_sdk::borsh::BorshSerialize`
   = note: this error originates in the derive macro `BorshSerialize` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `like::Like` with `#[derive(Ord)]`
```
For this struct:
```
#[derive(BorshSerialize, BorshDeserialize, Serialize, Deserialize, Clone)]
#[serde(crate = "near_sdk::serde")]
#[borsh(crate = "near_sdk::borsh")]
pub struct Post {
    pub id: PostId,
    pub author_id: AccountId,
    pub likes: HashSet<Like>,
    pub snapshot: PostSnapshot,
    // Excludes the current snapshot itself.
    pub snapshot_history: Vec<PostSnapshot>,
}

Reply:
Ah, I see, yes that makes sense, I forgot about that HashSet that needs to be borsh-serialized.